akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -63,7 +75,16 @@ private final class ServerSentEventParser(maxEventSize: Int) extends GraphStage[
       override def onPush() = {
         val (newLines, newSize) = grab(in) match {
           case "" => // A server-sent event is terminated with a new line, i.e. an empty line
-            emit(out, parseServerSentEvent(lines))
+            val (data, eventType, id, retry) = parseFields(lines)
+            if (data.isEmpty)
```

Comment:
Since you pull when there is no data, then you could make the parsing incremental rather than reparsing everything until there is data. I.e. have a mutable builder which you then apply lines to, as they arrive, and then have it create SSEs when there is something to send.